From f34f684103d16520fd22777377c630a947c7de0c Mon Sep 17 00:00:00 2001
From: David King <dking@redhat.com>
Date: Wed, 4 May 2016 19:43:52 +0100
Subject: [PATCH] Fix Add Files dialog button sensitivity

When the Add Files dialog has previously been opened and there are
selected files, the selection is restored for subsequent views of the
dialog. However, the sensitivity of the Add button is only updated when
manually selecting a file, as the dialog only updates the sensitivity
in response to the GtkCellRendererToggle for the selected state being
changed. When setting a list of selected files on the dialog, as done
when restoring the list of selected files on subsequent views of the
dialog, the Add button is only ever set to be insensitive.

Fix this by updating the Add button sensitivity after the list of files
has been loaded.

https://bugzilla.gnome.org/show_bug.cgi?id=738177
---
 src/fr-file-selector-dialog.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/fr-file-selector-dialog.c b/src/fr-file-selector-dialog.c
index 0e87b02..017df88 100644
--- a/src/fr-file-selector-dialog.c
+++ b/src/fr-file-selector-dialog.c
@@ -882,6 +882,7 @@ get_folder_content_done_cb (GError   *error,
 
 	gtk_tree_sortable_set_sort_column_id (GTK_TREE_SORTABLE (GET_WIDGET ("files_liststore")), sort_column_id, sort_order);
 	set_current_folder (self, load_data->folder);
+	_update_sensitivity (self);
 
 	if (load_data->dialog->priv->current_operation == load_data)
 		load_data->dialog->priv->current_operation = NULL;
-- 
2.8.2

